ci-automation/vendor-testing/gce.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+# Copyright (c) 2022 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+# Test execution script for the GCE vendor image.
+# This script is supposed to run in the mantle container.
+
+work_dir="$1"; shift
+arch="$1"; shift
+vernum="$1"; shift
+tapfile="$1"; shift
+
+# $@ now contains tests / test patterns to run
+
+source ci-automation/ci_automation_common.sh
+source sdk_lib/sdk_container_common.sh
+
+mkdir -p "${work_dir}"
+cd "${work_dir}"
+
+# We never run GCE on arm64, so for now fail it as an
+# unsupported option.
+if [[ "${arch}" == "arm64" ]]; then
+    echo "1..1" > "${tapfile}"
+    echo "not ok - all digital ocean tests" >> "${tapfile}"
+    echo "  ---" >> "${tapfile}"
+    echo "  ERROR: ARM64 tests not supported on GCE." | tee -a "${tapfile}"
+    echo "  ..." >> "${tapfile}"
+    exit 1
+fi
+
+channel="$(get_git_channel)"
+if [[ "${channel}" = 'developer' ]]; then
+    channel='alpha'
+fi
+testscript="$(basename "$0")"
+
+copy_from_buildcache "images/${arch}/${vernum}/${GCE_IMAGE_NAME}" .
+gcloud auth activate-service-account --key-file <(echo "${GCP_JSON_KEY}" | base64 --decode)
```

Comment:
This GCP_JSON_KEY decoding repeats 5 times - how about writing it out somewhere where it will be deleted afterwards?